pennylane/ops/qubit/matrix_ops.py

```diff
@@ -95,6 +95,30 @@ def _walsh_hadamard_transform(D: TensorLike, n: Optional[int] = None):
     return qml.math.reshape(qml.math.transpose(D), orig_shape)
 
 
+def _compute_udv(a, b):
+    r"""Given the matrices `a` and `b`, calculates the matrices `u`, `d` and `v`
+    of equation (36) in [arXiv-quant-ph:0504100](https://arxiv.org/pdf/quant-ph/0504100):
+
+    .. math::
+
+        a = u d v \\
+        b = u d^{\dagger} v.
+    """
+
+    # Calculates u and d diagonalizing ab^dagger (Eq.39)
+    ab_dagger = a @ qml.math.conj(b.T)
+    d_square, u = qml.math.linalg.eig(ab_dagger)
+    u, _ = qml.math.linalg.qr(u)
+
+    # complex square root of eigenvalues
```

Comment:
Why the complex square root? each of the diagonal entries should be real-valued no?